### PR TITLE
Explore: Fix deferred rendering of logs

### PR DESCRIPTION
**Pull request by aocenas**
Not exactly sure if by design or by error but the logs rendering was deferred for both first 100 lines and then another timeout for the rest. This first timeout was also a bit weird because the time was based on number of all rows. So if you had 1000 rows you wait 1s until we render the first 100 and then 2s to render the 900 rest.

### CODE REVIEW

File: packages/grafana-ui/src/components/Logs/LogRows.tsx [hunk 1]
```diff
@@ -138,7 +126,7 @@ class UnThemedLogRows extends PureComponent<Props, State> {
               onClickLabel={onClickLabel}
             />
           ))}
-        {hasData && deferLogs && <span>Rendering {rowCount} rows...</span>}
+        {hasData && !renderAll && <span>Rendering {rowCount - PREVIEW_LIMIT} rows...</span>}
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Logs/LogRows.tsx]: Found this, should be the same as above right?
```suggestion
        {hasData && !renderAll && <span>Rendering {rowCount - (previewLimit || PREVIEW_LIMIT)} rows...</span>}
```

File: packages/grafana-ui/src/components/Logs/LogRows.tsx [hunk 2]
```diff
@@ -22,45 +22,34 @@ export interface Props extends Themeable {
...
-    }
-  }
-
-  componentDidUpdate(prevProps: Props, prevState: State) {
-    // Staged rendering
-    if (prevState.deferLogs && !this.state.deferLogs && !this.state.renderAll) {
+    const { data, previewLimit } = this.props;
+    const rowCount = data ? data.rows.length : 0;
+    // Render all right away if not too far over the limit
+    const renderAll = rowCount <= (previewLimit || PREVIEW_LIMIT) * 2;
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Logs/LogRows.tsx]: Nitpick: might want to extract that into separate function and used it everywhere?
```suggestion
    const renderAll = rowCount <= this.getPreviewLimit() * 2;
```

**Review by hugohaggmark — APPROVED** [review]
LGTM! Left just some small comments that you could consider.

**Comment by aocenas** [comment]
@hugohaggmark yeah good point about the PREVIEW_LIMIT easy to forget it now, will change that.

**Comment by hugohaggmark** [comment]
> @hugohaggmark yeah good point about the PREVIEW_LIMIT easy to forget it now, will change that to use defaultProps.

Great solution with defaultProps!

**Comment by davkal** [comment]
> Not exactly sure if by design or by error but the logs rendering was deferred for both first 100 lines and then another timeout for the rest. This first timeout was also a bit weird because the time was based on number of all rows.

FTR this was a completely subjective tuning to give the browser some breathing room while/before processing/rendering tons. Seems snappy w/o it now 😄

**Pull request merged**